### PR TITLE
Be tolerant to string numbers in iframe resizing request.

### LINKED ISSUE #5608: requestResize gives wrong dimensions
RenderStart sets width of 300000 and height of 250000 by using values 300,250 on emulated desktop chrome

When I open an AMP page in actual desktop Chrome using emulation to iPhone and open e.g. [this](https://www.google.de/amp/www.bild.de/news/aktuelles/news-aktuell/zahl-der-hurrikantoten-in-haiti-steigt-auf-48282604,view=amp.bild.html) page, there is an adslot (first one) of 300x250.

In return of an ad global.context.renderStart is invoked using the dimensions 300x250 as you can see in the attached screenshot.

image

But in result the container gets the styles width: 300000px and height: 250000px;

image

### PR DESCRIPTION
**Pull request by lannka**
Refactored test code to reproduce the issue in test. The iframe should not be attached to document.body, but amp-ad element.

Fixes #5608

### CODE REVIEW

**Comment by jridgewell** [comment]
We should do the same for `<amp-iframe>`.

**Comment by lannka** [comment]
Good point. Will do it in a separate PR. Tracked here: #5801

File: extensions/amp-ad/0.1/test/test-amp-ad-xorigin-iframe-handler.js [hunk 1]
```diff
@@ -50,32 +56,26 @@ describe('amp-ad-xorigin-iframe-handler', () => {
...
   describe('init() returned promise', () => {
-    let iframe;
     let initPromise;
 
     describe('if render-start is implemented', () => {
 
       let noContentSpy;
 
-      beforeEach(() => {
+      beforeEach(done => {
```
Comment (zhouyx) [inline extensions/amp-ad/0.1/test/test-amp-ad-xorigin-iframe-handler.js]: So this `done` here is a special function that serves like a promise resolve func?

File: extensions/amp-ad/0.1/test/test-amp-ad-xorigin-iframe-handler.js [hunk 2]
```diff
@@ -39,9 +40,14 @@ describe('amp-ad-xorigin-iframe-handler', () => {
     const adElement = document.createElement('amp-ad');
     adElement.getAmpDoc = () => ampdoc;
     adImpl = new BaseElement(adElement);
+    document.body.appendChild(adElement);
     adImpl.uiHandler = new AmpAdUIHandler(adImpl);
     iframeHandler = new AmpAdXOriginIframeHandler(adImpl);
     testIndex++;
+
+    iframe = createIframeWithMessageStub(window);
```
Comment (zhouyx) [inline extensions/amp-ad/0.1/test/test-amp-ad-xorigin-iframe-handler.js]: I can't find where we append this iframe to DOM, will it load then?

File: extensions/amp-ad/0.1/test/test-amp-ad-xorigin-iframe-handler.js (hunk 1 above)
Comment (lannka) [inline extensions/amp-ad/0.1/test/test-amp-ad-xorigin-iframe-handler.js]: yes, take a look at [here](https://mochajs.org/), search for "ASYNCHRONOUS HOOKS"

File: extensions/amp-ad/0.1/test/test-amp-ad-xorigin-iframe-handler.js (hunk 2 above)
Comment (zhouyx) [inline extensions/amp-ad/0.1/test/test-amp-ad-xorigin-iframe-handler.js]: Ah! Good catch!

File: extensions/amp-ad/0.1/test/test-amp-ad-xorigin-iframe-handler.js (hunk 1 above)
Comment (jridgewell) [inline extensions/amp-ad/0.1/test/test-amp-ad-xorigin-iframe-handler.js]: It's a nodeback (an async callback that takes an error for the first parameter, if one occurred).

**Comment by lannka** [comment]
@zhouyx with the current `_ping_` implementation, there's no way to have an example in `fakead3p.amp.html` to demonstrate this. And I don't think there's need to change the implementation of `_ping_` for this particular purpose.

**Comment by zhouyx** [comment]
https://github.com/ampproject/amphtml/blob/master/test/manual/fakead3p.amp.html#L39
i think changing this line to `data-ad-width = '50px'`will work?

**Comment by lannka** [comment]
It will not, because we cast it to number in `_ping_.js`: https://github.com/ampproject/amphtml/blob/master/ads/_ping_.js#L42

**Comment by zhouyx** [comment]
I see. LGTM then.

**Comment by lannka** [comment]
Thanks for reviewing :-) !

**Pull request merged**